src/panels/alarm-configuration/alarmParser.tsx

```diff
@@ -0,0 +1,55 @@
+import { DataFrame, FieldCache, DataQueryRequest } from '@grafana/data';
+import { AlarmInfo, AlarmResultFields, AlarmQueryInfo } from './types';
+import { TwinMakerQuery } from 'common/manager';
+
+export function processAlarmQueryInput(query: DataQueryRequest<TwinMakerQuery> | undefined): AlarmQueryInfo {
+  const info: AlarmQueryInfo = {};
+
+  if (!query) {
+    info.invalidFormat = true;
+    info.warning = 'No query to parse';
+  }
+
+  const target = query?.targets[0];
+  info.alarmName = target?.componentName;
+  info.entityId = target?.entityId;
+
+  return info;
+}
+
+export function processAlarmResult(data: DataFrame[]): AlarmInfo {
```

Comment:
This is a mapping helper function, and it's highly view dependent, it should probably be a hook, or encapsulated into a component.